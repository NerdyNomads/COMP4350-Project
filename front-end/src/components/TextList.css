.text-list {
	background-color: #ffd89d;
	width: 60vw;
	position: absolute;
	right: 5vh;
	height: 95vh;
	color: black;
	padding-top: 5vh;
}
