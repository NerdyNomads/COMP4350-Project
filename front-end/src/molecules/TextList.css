.text-list {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    overflow: auto;
}