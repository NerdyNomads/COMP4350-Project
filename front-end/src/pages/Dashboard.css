.Dashboard {
    margin-left: 250px; /* size of sidebar */
    width: 100%; /* Fill the rest of the screen*/
}

.Dashboard-title {
    font-weight: 600;
    font-size: 36px;
    margin: 20px;
    display: flex;
    justify-content: space-between;
}

.Dashboard-title-reload {
    display: flex;
    justify-content: space-between;
    column-gap: 10px;
}