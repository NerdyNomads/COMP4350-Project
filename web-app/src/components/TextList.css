.text-list {
    background-color: #FFD89D;
    width: 60vw;
    position: absolute;
    right: 5vh;
    height: 95vh;
    color: black;
    padding-top: 5vh;
}