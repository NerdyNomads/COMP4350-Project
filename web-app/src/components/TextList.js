import React from "react";
import TextBox from "./TextBox";
class TextList extends React.Component {
    render() {
        return (
            <div className= "text-list">
            <b>You are currently viewing: Quotes</b>
            <TextBox/>
            <TextBox/>
            </div>
        )
    }
}

export default TextList;