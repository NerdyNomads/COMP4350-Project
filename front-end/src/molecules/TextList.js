import React from "react";
import PropTypes from "prop-types";

import TextBox from "../atoms/TextBox";
import TextBoxAdd from "../atoms/TextBoxAdd";
import "./TextList.css";

function TextList( {list} ) {

  const renderList = () => {
    return list.map( (l) => {
      return <TextBox key={l.id} text={l.text} source={l.source} id={l.id} />;
    });
  };

  return (
    <div className= "text-list">
      {renderList()}
      {<TextBoxAdd showInput={false}/>}
    </div>
  );
}

TextList.propTypes = {
  list: PropTypes.array.isRequired
};

export default TextList;