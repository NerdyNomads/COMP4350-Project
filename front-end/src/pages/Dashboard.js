import React, {useState, useEffect} from "react";

import axios from "axios";

import TextList from "../molecules/TextList";
import "./Dashboard.css";

function Dashboard() {
  const [ textItems, setTextItems ] = useState(null);

  /**
   * Grabs data from the db.
   * 
   * @returns 				returns data from the db. 
   */
  async function getTexts() {
    let result = await axios.get(`${process.env.REACT_APP_BACKEND_SERVER}/texts`);
    return result?.data;
  }

  useEffect(async () => {
    let texts = await getTexts();

    if (texts) {
      setTextItems(texts);
    }
  }, []);

  return (
    <div className="Dashboard">
      { textItems &&
          <TextList list={textItems}/>
      }
    </div>
  );
}


export default Dashboard;