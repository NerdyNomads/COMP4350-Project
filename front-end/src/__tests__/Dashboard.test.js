/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-unused-vars */
import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { shallow, configure } from "enzyme";
import { act } from "react-dom/test-utils";
import axios  from "axios";

import Dashboard from "../pages/Dashboard";
import TextList from "../molecules/TextList";

configure({ adapter: new Adapter() });  
jest.mock("axios");

const textItems = [
  { text: "Test1", source: "Test1" },
  { text: "Test1", source: "Test1" },
];

const whenStable = async () => {
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 0));
  });
};

test("Should match the snapshot.", () => { 
  let wrapper = shallow(<Dashboard workspaceId=""/>);
  expect(wrapper.html()).toMatchSnapshot();
});

// test("TextList component should be displayed when the data has been received.", async () => { 
//   axios.get.mockResolvedValueOnce({data: textItems});

//   let realUseState = React.useState;
//   let stubInitialState = textItems;
//   jest.spyOn(React, "useState").mockImplementationOnce(() => {
//     realUseState(stubInitialState);
//   });

//   let wrapper = mount(<Dashboard workspaceId=""/>);
//   await whenStable();
//   expect(axios.get).toHaveBeenCalled();

//   wrapper.update();
//   expect(wrapper.containsMatchingElement(<TextList/>)).toEqual(true);
// });

