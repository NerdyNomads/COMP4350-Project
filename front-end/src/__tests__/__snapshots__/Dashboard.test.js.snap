// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot. 1`] = `"<div class=\\"Dashboard\\"><div class=\\"Dashboard-title\\"></div></div>"`;
